File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int height;
int main(void)
{
    //checking userinput
    do
    {
        printf("Please input height: ");
        scanf("%i", &height);
    }
    while (height < 1 || height > 8);

    //iterating through height
    for (int i = 0; i < height; i++)
    {
        //iterating through height - i - 1 so im always adding 8 - the current hashes
        for (int space = 0; space < height - i - 1; space++)
        {
            printf(" ");
        }
        //iterate through height and printing hash tag
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        //printing the space between the left alligned stair case and the right alligned one
        printf("  ");
        //iterating same as j but here im not making it left alligned
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py


# equivelant of do while in python
height = 0
while height > 8 or height < 1:
    try:
        height = int(input("Height: "))
    except ValueError:
        height = int(input("Height: "))


# looping through height then doing the spacing and then printing the hash and new line and spacing then printing hash again
for i in range(height):
    # spacing
    for space in range(height - i - 1):
        print(" ", end="")
    # left alligned staircase
    for j in range(i + 1):
        print("#", end="")
    # space between both
    print("  ", end="")
    # right alligned staircase
    for k in range(i + 1):
        print("#", end="")
    print()
